### ISSUE TITLE
Replace filter: unescape backslashes

### ISSUE THREAD

**Issue by djbe** [SwiftGen/StencilSwiftKit#109]
In stencil, if I try:
`value|replace:"\n","\\n"`

The search and replace strings become `"\\n"` and `"\\\\n"` respectively. I think we should un-escape these backslashes, so that users correctly can search for newlines (for example). Or is this something that we should handle in Stencil itself?

**Comment by AliSoftware** [comment]
Mmmh good question. Is it:

 - The job of the `replace` filter itself
 - The job of Stencil itself (which would allow to fix that issue for all tags, but might be a breaking change too)
 - The job of SwiftGenKit (during parsing of the YAML/JSON/PLIST files)?

**Comment by djbe** [comment]
Note: SGK correctly parses the files (the string value is as it should be, unchanged).

**Comment by AliSoftware** [comment]
Yeah the parsing itself shouldn't be changed, but should we or should we not convert newlines to `\n` when we build the context from those parsed content, or is that the job for Stencil or the template, that's the real question :wink:

**Comment by djbe** [comment]
Keep in mind that if we do that in SGK, those strings may be somewhere quite deep in some array/dictionary structure, and we'd need to check all strings. Seems quite the overhead on the context builder, where it's honestly an issue in Stencil(SwiftKit).

**Comment by djbe** [comment]
Note: we also have a `contains` filter where this applies as well.

**Comment by AliSoftware** [comment]
Yeah which is why I'm starting to wonder if it shouldn't be Stencil's job to do that un-escape, instead of each filter individually…

**Comment by djbe** [comment]
Right, so the culprit would be this line:
https://github.com/stencilproject/Stencil/blob/521a599a60206b61b1ac76dd0f01d20f95d3d588/Sources/Variable.swift#L53

Anyone know what's the best way to unescape something like `"\\n"` or `"\\u5404\\u500b\\u90fd"` (we may as well do this for unicode sequences)?
